test/remote_test_data.go

```diff
@@ -0,0 +1,57 @@
+package test
+
+import (
+	"io/ioutil"
+	"os"
+	"path"
+	"runtime"
+)
+
+// TestDataProvider defines the simple funcion for returning data from a remote provider
+type TestDataProvider func() ([]byte, error)
+
+// LoadTestData attempt to load test data from local disk unless;
+// * It does not exist or,
+// * Variable REFRESH_DATA is present in ENV
+//
+// Data is stored under examples/test
+// This is done to avoid always depending on remote systems, but also with an option
+// to refresh/retest against the 'current' remote system data without manual copy/paste
+func LoadTestData(filename string, provider TestDataProvider) ([]byte, error) {
+	refreshLocalData := func(path string, refresh TestDataProvider) ([]byte, error) {
+		content, err := refresh()
+		if err != nil {
+			return nil, err
+		}
+		err = ioutil.WriteFile(path, content, 0644)
+		if err != nil {
+			return nil, err
+		}
+		return content, nil
+	}
+
+	// Get path to src/github.com/almighty/almighty-core/test/remote_test_data.go
+	_, packagefilename, _, ok := runtime.Caller(0)
+	if !ok {
+		panic("No caller information")
+	}
+
+	// The target dir would be src/github.com/almighty/almighty-core/examples/test
+	targetDir := path.Dir(packagefilename) + "/../examples/test/"
```

Comment:
This code will not work under Microsoft Windows as the path separator is Unix style.